### PR TITLE
added database file watching feature based on native fs watch

### PR DESCRIPTION
**Pull request by id0Sch**
Update README.md

### CODE REVIEW

**Comment by id0Sch** [comment]
@runk - do you plan on merging this anytime soon?

File: index.js [hunk 1]
```diff
@@ -15,12 +15,27 @@ exports.open = function(filepath, opts, cb) {
 
   fs.readFile(filepath, function(err, database) {
     if (err) cb(err);
-    else cb(null, new Reader(database, opts));
+    else {
+      var reader = new Reader(database, opts);
+      if (opts && !!opts.watchForUpdates) {
+        fs.watch(filepath, function() {
+          reader.load(fs.readFileSync(filepath));
```
Comment (runk) [inline index.js]: Can you use async api please

File: test/index.js [hunk 2]
```diff
@@ -57,9 +81,28 @@ describe('index', function() {
...
+        cb();
+      });
+      readFileSpy = sinon.spy(fs, 'readFileSync');
+      done();
+    });
+    afterEach(function(done) {
+      fs.watch.restore();
+      readFileSpy.restore();
+      done();
+    });
```
Comment (runk) [inline test/index.js]: It looks like you can reuse these test hooks.

File: test/index.js [hunk 3]
```diff
@@ -26,6 +27,19 @@ describe('index', function() {
   });
 
   describe('open()', function() {
+    var readFileSpy;
+    beforeEach(function(done) {
```
Comment (runk) [inline test/index.js]: Hooks are sync - there's no point using `done`

File: test/index.js (hunk 2 above)
Comment (id0Sch) [inline test/index.js]: i can, but i reinit' them after each test case so the count of (calledOnce / calledTwice) will be for each test and not aggregated.

File: index.js (hunk 1 above)
Comment (id0Sch) [inline index.js]: np

**Comment by id0Sch** [comment]
fixed all comments :)

**Comment by id0Sch** [comment]
hi @runk , do you think it's going to be merged anytime soon? i fixed all your comments
thanks man!

**Comment by runk** [comment]
Thanks for your work mate.

**Pull request merged**